tests/test_batch.py

```diff
@@ -217,5 +219,335 @@ def given_configuration_file_version_different_as_river_file_version_when_runnin
+            self.set_valid_general_section(config, "1.0")
+            config.set("General", "Qthreshold", "100")
+            
+            assert dfastmi.batch.check_configuration(rivers, config) is False
+            
+        def given_general_section_with_qthreshold_and_qbankfull_when_check_configuration_then_return_false(self, rivers : RiversObject, config : ConfigParser):
+            self.set_valid_general_section_with_q_values(config, "1.0")
+            
+            assert dfastmi.batch.check_configuration(rivers, config) is False
+            
+        def given_q_sections_with_discharge_check_configuration_then_return_false(self, rivers : RiversObject, config : ConfigParser):           
+            self.set_valid_general_section_with_q_values(config, "1.0")
+            self.add_q_section(config)
+            
+            assert dfastmi.batch.check_configuration(rivers, config) is False
+            
+        def given_mode_specific_test_with_discharge_check_configuration_then_return_true(self, rivers : RiversObject, config : ConfigParser):
+            self.set_valid_general_section_with_q_values(config, "1.0")
+            self.add_q_section(config)
+            config.set("General", "mode", "test")
+            
+            assert dfastmi.batch.check_configuration(rivers, config) is True
+            
+        def set_valid_general_section(self, config : ConfigParser, version : str):
+            config.add_section("General")
+            config.set("General", "Version", version)
+            config.set("General", "Branch", "Bovenrijn & Waal")
+            config.set("General", "Reach", 'Bovenrijn                    km  859-867')
+            
+        def set_valid_general_section_with_q_values(self, config : ConfigParser, version : str):
+            self.set_valid_general_section(config, version)
+            config.set("General", "Qthreshold", "100")
+            config.set("General", "Qbankfull", "100")
+            
+        def add_q_section(self, config : ConfigParser):
+            config.add_section("Q1")
+            config.set("Q1", "Discharge", "1300.0")
+            config.add_section("Q2")
+            config.set("Q2", "Discharge", "1300.0")
+            config.add_section("Q3")
+            config.set("Q3", "Discharge", "1300.0")
+    
+    class Test_check_configuration_v2():
+        @pytest.fixture
+        def rivers(self):
+            return  RiversObject("dfastmi/Dutch_rivers_v2.ini")
+        
+        @pytest.fixture
+        def config(self):
+            return  ConfigParser()
+        
+        def given_version_2_when_check_configuration_then_return_false(self, rivers : RiversObject, config : ConfigParser):
+            assert dfastmi.batch.check_configuration(rivers, config) is False
+        
+        def given_general_section_when_check_configuration_then_return_false(self, rivers : RiversObject, config : ConfigParser):
+            self.set_valid_general_section(config, "2.0")
```

Comment:
self.set_valid_general_section(config, "2.0") in test fixture setup?

Reply:
will update

Reply:
done